app/lib/features/item/cubit/item_cubit.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:async';
+
+import 'package:aev_sdk/aev_sdk.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:items_repository/items_repository.dart';
+import 'package:logger/logger.dart';
+import 'package:storages_repository/storages_repository.dart';
+
+part 'item_state.dart';
+
+class ItemCubit extends Cubit<ItemState> with LoggerMixin {
+  ItemCubit({
+    required ItemsRepository repo,
+    required ItemEntity item,
+  })  : _repo = repo,
+        super(ItemState.initial(item)) {
+    _itemsRepoSubscription = _repo.listen((_, repoState) {
+      if (repoState is ItemsRepositoryItemUpdatedSuccess) {
+        // if the expiration date is not the same, the inventory bloc updates
+        // the items list, so it is not necessary to update this cubit state.
+        if (repoState.item.uuid == state.item.uuid &&
+            repoState.item.expirationDate ==
+                repoState.prevItem.expirationDate &&
+            repoState.item.openedAt == repoState.prevItem.openedAt) {
+          _update(repoState.item);
+        }
+      }
+    });
+  }
+
+  final ItemsRepository _repo;
+
+  late final StreamSubscription<ObservableEvent<ItemsRepositoryState>>
+      _itemsRepoSubscription;
+
+  void _update(ItemEntity newItem) {
+    emit(state.copyWith(status: StateStatus.success, item: newItem));
+  }
+
+  Future<void> onOpened() async {
+    // emit(state.copyWith(status: StateStatus.progress));
```

Comment:
senza lo state "in progress" due state success di fila non vengono emessi

Reply:
in realtà potremmo non emettere stato tanto viene emesso dal listener l'update e la c'è anche l'item. Non potranno mai essere uguali 2 stati successivi.

Reply:
in realtà potremmo non emettere stato tanto viene emesso dal listener l'update e la c'è anche l'item. Non potranno mai essere uguali 2 stati successivi.